### PR TITLE
Update DRAGEN content and screenshot

### PR DESCRIPTION
**Pull request by justinegeffen**
Updates for https://github.com/seqeralabs/nf-tower-docs/issues/565

Addresses https://seqera.atlassian.net/browse/EDU-53

### CODE REVIEW

**Comment by justinegeffen** [comment]
@llewellyn-sl, I debated whether we should include the screenshot in the 23.3 version but as it was there anyway I kept it.

File: platform_versioned_docs/version-23.3/supported_software/dragen/_images/dragen_ce_ami.png
Comment (christopher-hakkaart) [inline platform_versioned_docs/version-23.3/supported_software/dragen/_images/dragen_ce_ami.png:1]: Why does the description under "Enable Dragon (Beta)" have single quotes around it?

Comment (christopher-hakkaart) [reply platform_versioned_docs/version-23.3/supported_software/dragen/_images/dragen_ce_ami.png:1]: Maybe it's a design thing?

Comment (christopher-hakkaart) [reply platform_versioned_docs/version-23.3/supported_software/dragen/_images/dragen_ce_ami.png:1]: Sweet!

**Review by christopher-hakkaart — APPROVED** [review]

**Pull request merged**